dbgd.c

```diff
@@ -30,6 +30,20 @@
 #define HTTPD_DEBUG LWIP_DBG_OFF
 #endif
 
+static void* get_transfer_buffer(uint32_t size) {
+    static uint32_t buffer_size = 0;
```

Comment:
Be aware using static variables kills multithreaded use.

Reply:
This was more of a workaround to avoid a leak and get this merged quickly.
I agree that we should eventually have an object for each connection.

I'll create an issue about improving this after merge. Thanks for pointing it out.